android/src/main/java/com/stripeterminalreactnative/listener/RNBluetoothReaderListener.kt

```diff
@@ -2,39 +2,17 @@ package com.stripeterminalreactnative.listener
 
 import com.facebook.react.bridge.ReactApplicationContext
 import com.stripe.stripeterminal.external.callable.Cancelable
-import com.stripe.stripeterminal.external.callable.ReaderListener
-import com.stripe.stripeterminal.external.models.BatteryStatus
-import com.stripe.stripeterminal.external.models.DisconnectReason
-import com.stripe.stripeterminal.external.models.ReaderEvent
-import com.stripe.stripeterminal.external.models.ReaderDisplayMessage
-import com.stripe.stripeterminal.external.models.ReaderInputOptions
-import com.stripe.stripeterminal.external.models.ReaderSoftwareUpdate
-import com.stripe.stripeterminal.external.models.TerminalException
+import com.stripe.stripeterminal.external.callable.MobileReaderListener
+import com.stripe.stripeterminal.external.callable.ReaderDisconnectListener
+import com.stripe.stripeterminal.external.models.*
+import com.stripeterminalreactnative.*
 import com.stripeterminalreactnative.ReactExtensions.sendEvent
-import com.stripeterminalreactnative.ReactNativeConstants.DISCONNECT
-import com.stripeterminalreactnative.ReactNativeConstants.FINISH_INSTALLING_UPDATE
-import com.stripeterminalreactnative.ReactNativeConstants.REPORT_AVAILABLE_UPDATE
-import com.stripeterminalreactnative.ReactNativeConstants.REPORT_UPDATE_PROGRESS
-import com.stripeterminalreactnative.ReactNativeConstants.REQUEST_READER_DISPLAY_MESSAGE
-import com.stripeterminalreactnative.ReactNativeConstants.REQUEST_READER_INPUT
-import com.stripeterminalreactnative.ReactNativeConstants.START_INSTALLING_UPDATE
-import com.stripeterminalreactnative.mapFromReaderDisconnectReason
-import com.stripeterminalreactnative.ReactNativeConstants.UPDATE_BATTERY_LEVEL
-import com.stripeterminalreactnative.ReactNativeConstants.REPORT_LOW_BATTERY_WARNING
-import com.stripeterminalreactnative.ReactNativeConstants.REPORT_READER_EVENT
-import com.stripeterminalreactnative.mapFromBatteryStatus
-import com.stripeterminalreactnative.mapFromReaderDisplayMessage
-import com.stripeterminalreactnative.mapFromReaderEvent
-import com.stripeterminalreactnative.mapFromReaderInputOptions
-import com.stripeterminalreactnative.mapFromReaderSoftwareUpdate
-import com.stripeterminalreactnative.nativeMapOf
-import com.stripeterminalreactnative.putError
-import com.stripeterminalreactnative.putDoubleOrNull
+import com.stripeterminalreactnative.ReactNativeConstants.*
 
 class RNBluetoothReaderListener(
     private val context: ReactApplicationContext,
     private val onStartInstallingUpdate: (cancelable: Cancelable?) -> Unit
-) : ReaderListener {
+) : MobileReaderListener, ReaderDisconnectListener by readerDisconnectDelete(context) {
```

Comment:
Out of curiosity, why are we adding a `readerDisconnectDelete` delegate here?

Reply:
To avoid adding same code again and again, as these interface is not required so people can remove if not carefully, for example the RNHandoffReaderListener is not implement the disconnect part before.